pkg/controller/scylladbcluster/sync.go

```diff
@@ -266,108 +256,163 @@ func (scc *Controller) sync(ctx context.Context, key string) error {
-		func() ([]metav1.Condition, error) {
-			return scc.syncServices(ctx, sc, remoteNamespaces, remoteControllers, remoteServiceMap, managingClusterDomain)
-		},
-	)
-	if err != nil {
-		errs = append(errs, fmt.Errorf("can't sync remote services: %w", err))
-	}
+		err = controllerhelpers.RunSync(
+			&status.Conditions,
+			fmt.Sprintf(remoteNamespaceControllerDatacenterProgressingConditionFormat, dc.Name),
+			fmt.Sprintf(remoteNamespaceControllerDatacenterDegradedConditionFormat, dc.Name),
+			sc.Generation,
+			func() ([]metav1.Condition, error) {
+				return scc.syncRemoteNamespaces(ctx, sc, &dc, remoteNamespaceMap[dc.RemoteKubernetesClusterName], managingClusterDomain)
+			},
+		)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("can't sync remote namespaces: %w", err))
+		}
 
-	err = controllerhelpers.RunSync(
-		&status.Conditions,
-		remoteEndpointSliceControllerProgressingCondition,
-		remoteEndpointSliceControllerDegradedCondition,
-		sc.Generation,
-		func() ([]metav1.Condition, error) {
-			return scc.syncEndpointSlices(ctx, sc, remoteNamespaces, remoteControllers, remoteEndpointSlicesMap, managingClusterDomain)
-		},
-	)
-	if err != nil {
-		errs = append(errs, fmt.Errorf("can't sync remote endpointslices: %w", err))
-	}
+		err = controllerhelpers.RunSync(
+			&status.Conditions,
+			fmt.Sprintf(remoteRemoteOwnerControllerDatacenterProgressingConditionFormat, dc.Name),
+			fmt.Sprintf(remoteRemoteOwnerControllerDatacenterDegradedConditionFormat, dc.Name),
+			sc.Generation,
+			func() ([]metav1.Condition, error) {
+				var progressingConditions []metav1.Condition
+
+				remoteNamespace := remoteNamespaces[dc.RemoteKubernetesClusterName]
+				if remoteNamespace == nil {
+					progressingConditions = append(progressingConditions, metav1.Condition{
+						Type:               fmt.Sprintf(remoteRemoteOwnerControllerDatacenterProgressingConditionFormat, dc.Name),
+						Status:             metav1.ConditionTrue,
+						Reason:             "WaitingForRemoteNamespace",
+						Message:            fmt.Sprintf("Waiting for Namespace to be created in %q Cluster", dc.RemoteKubernetesClusterName),
+						ObservedGeneration: sc.Generation,
+					})
+					return progressingConditions, nil
+				}
+
+				return scc.syncRemoteRemoteOwners(ctx, sc, &dc, remoteNamespace, remoteRemoteOwnerMap[dc.RemoteKubernetesClusterName], managingClusterDomain)
+			},
+		)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("can't sync remote remoteowners: %w", err))
+		}
 
-	err = controllerhelpers.RunSync(
-		&status.Conditions,
-		remoteEndpointsControllerProgressingCondition,
-		remoteEndpointsControllerDegradedCondition,
-		sc.Generation,
-		func() ([]metav1.Condition, error) {
-			return scc.syncEndpoints(ctx, sc, remoteNamespaces, remoteControllers, remoteEndpointsMap, managingClusterDomain)
-		},
-	)
-	if err != nil {
-		errs = append(errs, fmt.Errorf("can't sync remote endpoints: %w", err))
-	}
+		err = controllerhelpers.SyncRemoteNamespacedObject(
```

Comment:
Can this sequence of `SyncRemoteNamespacedObject` calls (where only arguments differ) be a loop?

Reply:
done